pairing/pairing.py

```diff
@@ -155,9 +154,18 @@ def _check_validity(c_I):
     Boolean 'True' or 'False'
     """
 
-    test_indirect = generate_indirect_connectivity(c_I)
+    test_indirect = _generate_indirect_connectivity(c_I)
     return (test_indirect == c_I).all()
 
+def new_generate_indirect(direct_corr):
+    new_indirect = _generate_indirect_connectivity(
+            direct_corr)
+    while _check_validity(new_indirect) == False:
```

Comment:
Nice! Have you seen any performance dip from this? I don't know if we have cases requiring multiple calls

Reply:
I'll run some tests to see